corehq/apps/auditcare/middleware.py

```diff
@@ -17,75 +17,37 @@ def __init__(self, get_response=None):
                     "If you want to setup central view call audit events, "
                     "please add the property and populate it with fully "
                     "qualified view names."
                 )
-            elif not hasattr(settings, "AUDIT_MODULES"):
+            elif not self.audit_modules:
                 log.warning(
                     "You do not have the AUDIT_MODULES settings variable "
                     "setup. If you want to setup central module audit events, "
                     "please add the property and populate it with module names."
                 )
 
-            if hasattr(settings, "AUDIT_VIEWS") or hasattr(settings, "AUDIT_MODULES"):
-                self.active = True
-            else:
-                self.active = False
-        else:
-            self.active = True
-
-    @staticmethod
-    def do_process_view(request, view_func, view_args, view_kwargs, extra={}):
-        if (
-            getattr(settings, 'AUDIT_MODULES', False)
-            or getattr(settings, 'AUDIT_ALL_VIEWS', False)
-            or getattr(settings, "AUDIT_VIEWS", False)
-        ):
-
-            if isinstance(view_func, FunctionType):
-                fqview = "%s.%s" % (view_func.__module__, view_func.__name__)
-            else:
-                #just assess it from the classname for the class based view
-                fqview = "%s.%s" % (view_func.__module__, view_func.__class__.__name__)
-            if fqview == "django.contrib.staticfiles.views.serve" or fqview == "debug_toolbar.views.debug_media":
-                return None
-
-            def check_modules(view, audit_modules):
-                return any((view.startswith(m) for m in audit_modules))
-
-            audit_doc = None
-
-            if (fqview.startswith('django.contrib.admin')
-                    or fqview.startswith('reversion.admin')) and getattr(settings, "AUDIT_ADMIN_VIEWS", True):
-                audit_doc = AuditEvent.audit_view(request, request.user, view_func, view_kwargs)
-            elif (
-                check_modules(fqview, settings.AUDIT_MODULES)
-                or fqview in settings.AUDIT_VIEWS
-                or getattr(settings, 'AUDIT_ALL_VIEWS', False)
-            ):
-                audit_doc = AuditEvent.audit_view(request, request.user, view_func, view_kwargs, extra=extra)
-            if audit_doc:
-                setattr(request, 'audit_doc', audit_doc)
-        return None
-
     def process_view(self, request, view_func, view_args, view_kwargs):
         """
         Simple centralized manner to audit views without having to modify the
         requisite code.  This is an alternate way to manage audit events rather
         than using the decorator.
         """
-        AuditMiddleware.do_process_view(request, view_func, view_args, view_kwargs)
+        if self.active and self.should_audit(view_func):
+            audit_doc = NavigationEventAudit.audit_view(request, request.user, view_func, view_kwargs)
```

Comment:
Trust it's correct, but why did this change from `AuditEvent.audit_view` to `NavigationEventAudit.audit_view`?

Reply:
nvm, found where it's monkey-patched. bizarre.